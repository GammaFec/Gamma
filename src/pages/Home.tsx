import React from "react";

const Home = (): JSX.Element => <div>Home</div>;

export default Home;
