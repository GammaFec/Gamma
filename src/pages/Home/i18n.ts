import { LanguageList } from "./../../i18next/config";
import i18next from "i18next";

i18next.addResources(LanguageList.SPANISH, "Home", {
    Title: "Titulo",
    Paragraph:
        "Español Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam finibus dictum est, vel tincidunt enim iaculis nec. Cras vitae placerat nunc. Maecenas eu neque ac enim ullamcorper condimentum ut id ex. Donec porta vel justo at interdum. Curabitur vitae diam leo. Pellentesque a efficitur nulla. Suspendisse facilisis eu enim eu tempor. Fusce erat metus, tempor quis lacus nec, accumsan ornare erat. Sed viverra et tellus at mattis.",
    SignUp: "Registarse",
    LogIn: "Iniciar sesión",
    ImageAlt: "Imagen de un perrx o gatx"
});

i18next.addResources(LanguageList.ENGLISH, "Home", {
    Title: "Title",
    Paragraph:
        "English Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam finibus dictum est, vel tincidunt enim iaculis nec. Cras vitae placerat nunc. Maecenas eu neque ac enim ullamcorper condimentum ut id ex. Donec porta vel justo at interdum. Curabitur vitae diam leo. Pellentesque a efficitur nulla. Suspendisse facilisis eu enim eu tempor. Fusce erat metus, tempor quis lacus nec, accumsan ornare erat. Sed viverra et tellus at mattis.",
    SignUp: "Sign up",
    LogIn: "Log in",
    ImageAlt: "Dog or Cat Image"
});
