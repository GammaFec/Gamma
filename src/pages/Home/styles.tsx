import styled from "styled-components";
import { fontSizes, theme } from "../../common/styles/index";

const StyledHome = styled.section`
    font-family: ${theme.fontPrimary};
    box-sizing: border-box;
    width: 100%;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    padding: 69px 16px;
`;

const StyledContSup = styled.header`
    max-width: 250px;
    text-align: center;
    box-sizing: border-box;
    width: 80vw;
`;

const StyledContInf = styled.div`
    box-sizing: border-box;
    width: 80vw;
    height: 148px;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
`;

const StyledImg = styled.img`
    width: 110.76px;
`;

const StyledH1 = styled.h1`
    font-size: ${fontSizes.font21};
    font-weight: 600;
    line-height: 31.5px;
    letter-spacing: -0.44px;
`;

export { StyledHome, StyledContSup, StyledContInf, StyledImg, StyledH1 };
