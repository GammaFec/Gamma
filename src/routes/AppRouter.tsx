import React, { Suspense } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Layout from "../components/Layout";
import Splash from "../pages/Splash";
import PrivateRoute from "./PrivateRoute";

const Login = React.lazy(() => import("../pages/Login"));
const index = React.lazy(() => import("../pages/HomeScreen"));

const AppRouter = () => {
    return (
        <>
            <Suspense fallback={Splash}>
                <Router>
                    <Switch>
                        <Route component={Login} path="/login" />
                        <Route path="/homescreen" component={index} />
                        <PrivateRoute path="/">
                            <Layout />
                        </PrivateRoute>
                    </Switch>
                </Router>
            </Suspense>
        </>
    );
};

export default AppRouter;
