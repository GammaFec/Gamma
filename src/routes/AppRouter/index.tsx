import React, { Suspense } from "react";
import { BrowserRouter as Router, Switch } from "react-router-dom";

import RouteItem from "../RouteItem";
import ROUTES from "../routes";

import Splash from "../../pages/Splash";

const AppRouter = (): JSX.Element => {
    return (
        <>
            <Suspense fallback={Splash}>
                <Router>
                    <Switch>
                        {ROUTES.map((route) => (
                            <RouteItem
                                component={route.component}
                                exact={route.exact && route.exact}
                                key={route.name}
                                path={route.path}
                            />
                        ))}
                    </Switch>
                </Router>
            </Suspense>
        </>
    );
};

export default AppRouter;
