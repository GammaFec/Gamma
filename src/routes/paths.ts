export enum PATHS {
    Splash = "/Splash",
    Home = "/home",
    Login = "/login",
    Messages = "/messages",
    MovilAuth = "/movil-auth/:id",
    Onboarding = "/onboarding"
}
