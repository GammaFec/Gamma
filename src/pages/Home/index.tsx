import React, { ReactElement } from "react";

//Translate - i18next
import { useTranslation } from "react-i18next";

// Styled-components
import { StyledHome, StyledHeader, StyledMain, StyledImg, StyledH1 } from "./styles";

// Components
import Logo from "../../assets/img/Logo.svg";
import Button from "../../components/Button";
import { PRIMARY, SECONDARY } from "../../common/styles/constants";
import { useHistory } from "react-router-dom";

const LoginMain = (): ReactElement => {
    const history = useHistory();
    const { t } = useTranslation("Home");

    const handleGoToMovilAuth = (): void => {
        history.push("/movil-auth/:id");
    };

    const handleGoToLogin = (): void => {
        history.push("/login");
    };

    return (
        <StyledHome>
            <StyledHeader>
                <StyledImg alt={t("ImageAlt")} src={Logo} />
                <StyledH1>{t("Title")}</StyledH1>
            </StyledHeader>
            <StyledMain>
                <Button handleClick={handleGoToMovilAuth} variant={PRIMARY}>
                    {t("SignUp")}
                </Button>
                <Button handleClick={handleGoToLogin} variant={SECONDARY}>
                    {t("LogIn")}
                </Button>
            </StyledMain>
        </StyledHome>
    );
};

export default LoginMain;
