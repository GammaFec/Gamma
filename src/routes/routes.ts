import React from "react";
import { PATHS } from "./paths";
import IRoute from "./types";

const Splash = React.lazy(() => import("../pages/Splash"));
const NotFound = React.lazy(() => import("../pages/NotFound"));
const Login = React.lazy(() => import("../pages/Login"));
const Home = React.lazy(() => import("../pages/Home"));

const ROUTES: IRoute[] = [
    {
        path: PATHS.Splash,
        component: Splash,
        name: "Splash",
        exact: true,
        isPrivate: false
    },
    {
        path: PATHS.Login,
        component: Login,
        name: "Login",
        exact: true,
        isPrivate: false
    },
    {
        path: PATHS.Home,
        component: Home,
        name: "Home",
        exact: true,
        isPrivate: false
    },
    {
        component: NotFound,
        name: "NotFound"
    }
];

export default ROUTES;
