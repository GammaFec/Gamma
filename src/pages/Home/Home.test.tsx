import React from "react";
import Home from "./";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { createMemoryHistory } from "history";
import { Router } from "react-router";
import { PATHS } from "../../routes/paths";
import "jest-styled-components";

jest.mock("react-i18next", () => ({
    useTranslation: (): { t: (key: string) => string } => ({ t: (key: string): string => key })
}));

describe("<Home />", () => {
    const history = createMemoryHistory();

    beforeEach(() => {
        history.push(PATHS.Home);
    });

    test("Render", () => {
        const { asFragment } = render(<Home />);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Render all childrens", () => {
        const { getByAltText, getByText } = render(<Home />);

        getByAltText("ImageAlt");
        getByText("Title");
        getByText("SignUp");
        getByText("LogIn");
    });

    test("Handle go to Login", () => {
        const { getByText } = render(
            <Router history={history}>
                <Home />
            </Router>
        );

        const Button = getByText("LogIn");

        fireEvent.click(Button);

        expect(history.location.pathname).toEqual(PATHS.Login);
    });

    test("Handle go to SignUp", () => {
        const { getByText } = render(
            <Router history={history}>
                <Home />
            </Router>
        );

        const Button = getByText("SignUp");

        fireEvent.click(Button);

        expect(history.location.pathname).toEqual(PATHS.MovilAuth);
    });
});
