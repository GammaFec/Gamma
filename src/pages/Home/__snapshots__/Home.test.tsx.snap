// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Render 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Poppins','sans-serif';
  width: 100%;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 69px 16px;
}

.c1 {
  max-width: 250px;
  text-align: center;
  width: 80vw;
}

.c4 {
  width: 80vw;
  height: 148px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 110.76px;
}

.c3 {
  font-size: 21px;
  font-weight: 600;
  line-height: 31.5px;
  -webkit-letter-spacing: -0.44px;
  -moz-letter-spacing: -0.44px;
  -ms-letter-spacing: -0.44px;
  letter-spacing: -0.44px;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  font-family: 'Poppins','sans-serif';
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.7rem 3rem;
  margin: 0.2rem;
  min-width: 243px;
  height: 48px;
  cursor: pointer;
  background-color: #FF7B28;
  border: #FF7B28 solid 2px;
  color: #FFFFFF;
}

.c5:hover {
  opacity: 0.6;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  font-family: 'Poppins','sans-serif';
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.7rem 3rem;
  margin: 0.2rem;
  min-width: 243px;
  height: 48px;
  cursor: pointer;
  background-color: #FFFFFF;
  border: #00A59E solid 2px;
  color: #00A59E;
}

.c6:hover {
  opacity: 0.6;
}

<section
    class="c0"
  >
    <header
      class="c1"
    >
      <img
        alt="ImageAlt"
        class="c2"
        src="Logo.svg"
      />
      <h1
        class="c3"
      >
        Title
      </h1>
    </header>
    <div
      class="c4"
    >
      <button
        class="c5"
      >
        SignUp
      </button>
      <button
        class="c6"
      >
        LogIn
      </button>
    </div>
  </section>
</DocumentFragment>
`;
